### PR TITLE
added local run guide and generating key material

### PR DESCRIPTION
**Pull request by pimg**

### CODE REVIEW

File: SETUP.md [hunk 1]
```diff
@@ -94,8 +94,70 @@ Build and run the docker containers
...
+This directory also includes a set of test keys, certificates and keystores. However, it is certainly possible to use your own key material.
+When using/runninng the reference implementation other than for local development it is highly recommended using your own key material.
+
+### Generating keystore for TLS
+The TLS connection is setup to require a PKCS12 keystore containing a private key and public certificate.
+To generate a self-signed certificate:
+```shell
+keytool -genkeypair -alias springboot-https -keyalg RSA -storetype PKCS12 -keystore springboot-https.p12 -storepass your_key-store_password
+```
+To use an existing TLS private key and certificate chain:
```
Comment (nt-as) [inline SETUP.md:130]: Should probably explicitly state "in PEM format" somewhere (We are trying to say "This is how to generate the P12 file we expect **if** you have the keys **in PEM format**").  In my experience, PEM and P12 are the two "big contenders" (With P12/PKCS12 being used more often in Windows environment and PEM being using more often in Linux).

File: SETUP.md [hunk 2]
```diff
@@ -94,8 +94,70 @@ Build and run the docker containers
...
+```shell
+keytool -genkeypair -alias dcsa-kid -keyalg RSA -keystore dcsa-jwk.jks -storepass dcsa-pass
+```
+
+### Generating keystore for verifying signatures
+Verifying the signatures of either incoming transferblocks or verifying the signature in the response of an outgoing transferblock requires a JKS.
+This JKS can contain multiple public certificates of each platform connected. The reference implementations uses the **_'CN'_** in the certificate to determine which public key to use for verification.
+For incoming transferblocks the CN is matched with the 'platformHost' field in the transaction.
+For verifying responses of outgoing transferblocks the CN is matched to the URL the transferblock is being sent to.
+So when generating the key material make sure to take not of the CN used.
```
Comment (nt-as) [inline SETUP.md:148]: Possible typo: "take **note** of ..."?

File: SETUP.md (hunk 1 above)
Comment (pimg) [reply SETUP.md:130]: PKCS12 is a keystore format whereas PEM is a format (base64 encoded) for keys and certificates

Comment (pimg) [reply SETUP.md]: fixed

File: SETUP.md (hunk 2 above)
Comment (pimg) [reply SETUP.md]: fixed

**Review by nt-as — APPROVED** [review]

**Pull request merged**